internal/resolve/catalog_test.go

```diff
@@ -963,3 +963,150 @@ func TestMultipleChannels(t *testing.T) {
 	assert.Equal(t, bsemver.MustParse("2.0.0"), *gotVersion)
 	assert.Equal(t, ptr.To(packageDeprecation(pkgName)), gotDeprecation)
 }
+
+func TestAllCatalogsDisabled(t *testing.T) {
+	pkgName := randPkg()
+	listCatalogs := func(ctx context.Context, options ...client.ListOption) ([]catalogd.ClusterCatalog, error) {
+		return []catalogd.ClusterCatalog{
+			{
+				Spec: catalogd.ClusterCatalogSpec{
+					Availability: "Disabled",
+				},
+			},
+			{
+				Spec: catalogd.ClusterCatalogSpec{
+					Availability: "Disabled",
+				},
+			},
+		}, nil
+	}
+
+	getPackage := func(ctx context.Context, cat *catalogd.ClusterCatalog, packageName string) (*declcfg.DeclarativeConfig, error) {
+		return genPackage(pkgName), nil
+	}
+
+	r := CatalogResolver{
+		WalkCatalogsFunc: CatalogWalker(listCatalogs, getPackage),
+	}
+
+	ce := buildFooClusterExtension(pkgName, []string{}, ">=1.0.0", ocv1alpha1.UpgradeConstraintPolicyCatalogProvided)
+	_, _, _, err := r.Resolve(context.Background(), ce, nil)
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "no enabled catalogs found for package")
+}
+
+func TestSomeCatalogsDisabled(t *testing.T) {
+	pkgName := randPkg()
+	listCatalogs := func(ctx context.Context, options ...client.ListOption) ([]catalogd.ClusterCatalog, error) {
+		return []catalogd.ClusterCatalog{
+			{
+				Spec: catalogd.ClusterCatalogSpec{
+					Availability: "Enabled",
+				},
+			},
+			{
+				Spec: catalogd.ClusterCatalogSpec{
+					Availability: "Disabled",
+				},
+			},
+		}, nil
+	}
+
+	getPackage := func(ctx context.Context, cat *catalogd.ClusterCatalog, packageName string) (*declcfg.DeclarativeConfig, error) {
+		return genPackage(pkgName), nil
+	}
+
+	r := CatalogResolver{
+		WalkCatalogsFunc: CatalogWalker(listCatalogs, getPackage),
+	}
+
+	ce := buildFooClusterExtension(pkgName, []string{}, ">=1.0.0", ocv1alpha1.UpgradeConstraintPolicyCatalogProvided)
+	gotBundle, gotVersion, _, err := r.Resolve(context.Background(), ce, nil)
+	require.NoError(t, err)
+	require.NotNil(t, gotBundle)
+	require.Equal(t, bsemver.MustParse("3.0.0"), *gotVersion)
+}
+
+func TestPriorityRespectedWithDisabledCatalogs(t *testing.T) {
```

Comment:
This test seems unnecessary to me since priority and availability are orthogonal concepts (right?)

We ignore the disabled catalogs in the `CatalogWalker`, so the resolver (where we consider priority) never even knows of the existence of the disabled catalog.

Reply:
modifying the test so it now effectively verifies that disabled catalogs are ignored even when they have higher priority and versions. If the logic to skip disabled catalogs breaks, the test will fail either by panic or by selecting the wrong bundle.